lib/widgets/mapView_widget.dart

```diff
@@ -0,0 +1,179 @@
+        child: FloatingActionButton(
+          elevation: 8,
+          //isExtended: true,
+          child: Icon(
+            Icons.location_on,
+            size: 36,
+            color: AppColors.accentRed,
+          ),
+          backgroundColor: AppColors.primaryWhite,
+          onPressed: () async {
+            // TODO: use streams for faster gps access
+            var position = await determineGPSPosition().catchError((onError) {
+              print(onError);
+            });
+            print(
+              'Szukam sygnału GPS... longitude: ' +
+                  position.longitude.toString() +
+                  ', latitude: ' +
+                  position.latitude.toString(),
+            );
+
+            // center and animate the map view on the item icon
+            var start = controller.value;
+            var end = controller.getCenteredViewMatrixGPS(
+              position.longitude,
+              position.latitude,
+              mapDataController.getMapGPSMinX(),
+              mapDataController.getMapGPSMinY(),
+              mapDataController.getMapGPSMaxX(),
+              mapDataController.getMapGPSMaxY(),
+              mapDataController.getMapGPSMapSizeX(),
+              mapDataController.getMapGPSMapSizeY(),
+            );
+            mapDataController.setupMapAnimation(controller);
+            mapDataController.navigateFromToPoint(start: start, end: end);
+            mapDataController.setCurrentMapIconIndex(0);
+
+            ScaffoldMessenger.of(context).removeCurrentSnackBar();
+            ScaffoldMessenger.of(context).showSnackBar(
+              SnackBar(
+                content: Text(
+                  'Jesteś tutaj... longitude: ' +
+                      position.longitude.toString() +
+                      ', latitude: ' +
+                      position.latitude.toString(),
+                  style: TextStyle(color: AppColors.primaryWhite),
+                ),
+                backgroundColor: AppColors.primaryNormal,
+                behavior: SnackBarBehavior.fixed,
+                duration: Duration(seconds: 3),
+              ),
+            );
+          },
+        ),
+      ),
+      body: Stack(
+        children: [
+          InteractiveViewer(
+            constrained: false,
+            transformationController: controller,
+            child: Stack(
+              children: List<Widget>.generate(1,
+                      (int index) => Image.asset("assets/images/wsd_map.png")) +
+                  List<Widget>.generate(
+                    mapDataController.getLength(),
+                    (int index) => MapPoint(
+                      mapTransformController: controller,
+                      mapItemIndex: index,
+                      mapDataController: mapDataController,
+                      scrollController: _scrollController,
+                    ),
+                    // TODO: add this segment to display icon for present gps position
```

Comment:
In my case the commented code, when used, throws an error. Shall we write a separate story to add a current location pin?

Reply:
The current solution is rather basic - proof of concept. I would opt for a new story to iron out all cases including:

- prompting for turning on location,
- out of bounds display,
- error checking (ex. cannot read precise gps coordinates)